lisp/ess-inf.el

```diff
@@ -250,18 +241,18 @@ This may be useful for debugging."
      ((and (not (comint-check-proc (current-buffer)))
            (derived-mode-p 'inferior-ess-mode))
       (current-buffer))
-     ;; Take the *R:N* buffer if already exists (and contains dead proc!)
-     ;; fixme: buffer name might have been changed, iterate over all
-     ;; inferior-ess buffers.
```

Comment:
This `fixme:` said something useful. Was that issue addressed?

Reply:
oops. I don't understand this fixme note so I don't know if it's still relevant, I'll add it back.

Reply:
It's terse and it's highly relevant to our other thread. The idea is that if there was a process "R:2" and it's buffer name was change from `*R:2*` to `*G:2*` manually (rename-buffer) and the process died, then this logic will not detect that buffer. Consequently the process restart breaks if buffer name changed (when called from the script buffer).

Would be great if it could be fixed now as you are on it. BTW, instead of iterating over all buffers I believe we can iterate over all our processes.

Reply:
This seems complicated, especially with the name generating function. If the user renamed `R:2` to `G:2`, it's not obvious to me that we should recycle that buffer. I would expect we don't actually.

Reply:
I think we should. At the very least for the restarts. If I want to restart the process in G:2 buffer, it should stay there. 

Also for consistency. If you start a new R session from an non-proc G:2 buffer, it will reuse the G:2 buffer.

Reply:
Agreed that reload should keep the same buffer. Let's keep the current logic and think about inferior restart as part of #871.